### PR TITLE
Replace buggy ace-editor with codemiror

### LINKED ISSUE #486: I can only write numbers in the console from mobile
As the title says, only numeric characters go through in the console when using my phone. I was doing a playground exercise, and not only could I not type, I can't paste copied code to test it either, so I had no choice but skip it.

### LINKED ISSUE #487: Shift key behaves as backspace on mobile
Happened to me on my Nexus 7 tablet, running  Android 6.0.1.

Perhaps it's an Ace bug, but makes Atheneum unusable. :cry:

### LINKED ISSUE #811: Full screen editor
We need an additional button to increase the size of the code editor, since it is difficult to use when solving large problems.

### PR DESCRIPTION
**Pull request by aguspina**
Fixes famous shift key ace editor bug by replacing ace editor 🤕  .

Some considerations:
* Gobstones mode is not supported
* Prolog mode is not supported
* after updating modules/_highlight.scss in mumuki-styles, this file should be deleted from this repo
* Also, I didn't check all highlight styles

Fixes #811
Fixes #487 
Fixes #486 
Related #813

### CODE REVIEW

File: app/assets/javascripts/application/codemirror-builder.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        mumuki.editor.setOption('mode', language);
+      }
+    },
+    setupPlaceholder: function (text) {
+      this.editor.setOption('placeholder', text);
+    },
+
+    setupOptions: function (minLines) {
+      this.editor.setOption('tabSize', 2);
+      this.editor.setOption('indentWithTabs', true);
```
Comment (fedescarpa) [inline app/assets/javascripts/application/codemirror-builder.js]: I prefer indent using spaces

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -1,3 +1,7 @@
+/*
+*= require codemirror
+*= require codemirror/addons/display/fullscreen
+*/
```
Comment (fedescarpa) [inline app/assets/stylesheets/application.scss]: it should be replaced with
```scss
@import 'codemirror';
@import 'codemirror/addons/display/fullscreen';
```

File: app/assets/stylesheets/application/modules/_highlight.scss [hunk 3]
```diff
@@ -4,48 +4,37 @@ pre {
...
-.ace_operator,
-.ace_meta.ace_rule.ace_definition.ace_prolog {
+.cm-operator {
   color: $brand-complementary !important;
   font-weight: bold !important;
 }
 
 /* comment */
-.ace_comment, .c, .cm, .cp, .c1, .cs {
+.cm-comment {
```
Comment (fedescarpa) [inline app/assets/stylesheets/application/modules/_highlight.scss]: I think we add code mirror clases without remove ace-editor and other clases. And make styles backward compatible.

However, I think this styles are duplicated in mumuki/mumuki-styles

File: app/assets/javascripts/application.js [hunk 4]
```diff
@@ -16,12 +16,17 @@
...
-//= require ace/mode-ruby
-//= require ace/mode-haskell
-//= require ace/mode-prolog
-//= require ace/mode-javascript
-//= require ace/mode-markdown
+//= require codemirror
+//= require codemirror/modes/ruby
+//= require codemirror/modes/haskell
+//= require codemirror/modes/javascript
+//= require codemirror/modes/markdown
```
Comment (flbulgarelli) [inline app/assets/javascripts/application.js]: Why markdown?

File: app/assets/javascripts/application.js [hunk 5]
```diff
@@ -16,12 +16,17 @@
...
 //= require bootstrap
-//= require ace-rails-ap
-//= require ace/mode-ruby
-//= require ace/mode-haskell
-//= require ace/mode-prolog
-//= require ace/mode-javascript
-//= require ace/mode-markdown
+//= require codemirror
+//= require codemirror/modes/ruby
+//= require codemirror/modes/haskell
```
Comment (flbulgarelli) [inline app/assets/javascripts/application.js]: What about other supported languages like C++, C, HTML, SQL, Python and Java?

**Comment by flbulgarelli** [comment]
Also, could we expose a simple API in the editor to allow adding in-line errors dynamically?

Something like this:

```javascript
editor.highlightErrorOnLine(lineNumber, text);
editor.highlightWarningOnLine(lineNumber, text);
```

**Pull request merged**